lib/src/facts/linux/fips_resolver.cc

```diff
@@ -0,0 +1,35 @@
+#include <internal/facts/linux/fips_resolver.hpp>
+#include <leatherman/file_util/file.hpp>
+#include <boost/algorithm/string.hpp>
+#include <boost/lexical_cast.hpp>
+
+using namespace std;
+
+using boost::lexical_cast;
+using boost::bad_lexical_cast;
+
+namespace lth_file = leatherman::file_util;
+
+namespace facter { namespace facts { namespace linux {
+
+    fips_resolver::data fips_resolver::collect_data(collection& facts)
+    {
+        data result;
+
+        // Below file might not exist in which case we need to ensure to not report
+        // anything incorrectly by initializing the result to a safe value
+        result.is_fips_mode_enabled = false;
```

Comment:
Might be better to specify a way in the data to check whether it is valid or not before adding the fact? I am not familiar with the FIPS mode, but as a user if I read "is_fips_mode_enabled = false" I might be lead to think that FIPS mode is not enabled when it in fact might be, but isn't reported in the /proc/sys/crypto/fips_enabled.

Reply:
Hmm... not sure. Either absence or a false value should be synonymous but don't mind not reporting the fact if not enabled.